nautobot/dcim/models/device_components.py

```diff
@@ -727,13 +727,17 @@ def clean(self):
                     )
 
         # Validate untagged VLAN
-        # TODO: after Location model replaced Site, which was not a hierarchical model, should we allow users to assign a VLAN belongs to
-        # the parent Locations or the child locations of `device.location`?
+        location = self.parent.location
```

Comment:
`self.parent` might be `None`, so perhaps:

```suggestion
        location = self.parent.location if self.parent is not None else None
```